### PR TITLE
iris

### PR DESCRIPTION
**Pull request by ocefpaf**
All the dependencies should be up now.

### CODE REVIEW

**Comment by pelson** [comment]
OSX: 

```
copying lib/iris/etc/palette/sequential/YlOrRd_09.txt -> build/lib.macosx-10.5-x86_64-3.4/iris/etc/palette/sequential
copying lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb -> build/lib.macosx-10.5-x86_64-3.4/iris/fileformats/_pyke_rules
/Users/travis/miniconda/envs/_build/bin/python tools/generate_std_names.py etc/cf-standard-name-table.xml build/lib.macosx-10.5-x86_64-3.4/iris/std_names.py
writing [iris.fileformats._pyke_rules.compiled_krb]/fc_rules_cf_fc.py
writing [iris.fileformats._pyke_rules.compiled_krb]/compiled_pyke_files.py
Traceback (most recent call last):
  File "setup.py", line 229, in <module>
    'clean_source': CleanSource},
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/site-packages/setuptools-20.6.7-py3.4.egg/setuptools/command/install.py", line 61, in run
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/distutils/command/install.py", line 539, in run
    self.run_command('build')
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/distutils/command/build.py", line 126, in run
    self.run_command(cmd_name)
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "setup.py", line 180, in run
    MakePykeRules._pyke_rule_compile()
  File "setup.py", line 134, in _pyke_rule_compile
    knowledge_engine.engine(iris.fileformats._pyke_rules)
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/site-packages/pyke/knowledge_engine.py", line 110, in __init__
    target_package.load(self, **kws)
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/site-packages/pyke/target_pkg.py", line 366, in load
    self.do_by_ext('load', target_filename, engine, load_flags)
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/site-packages/pyke/target_pkg.py", line 265, in do_by_ext
    return getattr(self, "%s_%s" % (prefix, ext))(filename, *args)
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/site-packages/pyke/target_pkg.py", line 374, in load_py
    self.load_module(module_path, target_filename, engine)
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/site-packages/pyke/target_pkg.py", line 407, in load_module
    module = import_(module_path)
  File "/Users/travis/miniconda/envs/_build/lib/python3.4/site-packages/pyke/target_pkg.py", line 463, in import_
    mod = __import__(modulename)
ImportError: No module named 'iris.fileformats._pyke_rules.compiled_krb'
Command failed: /bin/bash -x -e /Users/travis/build/conda-forge/staged-recipes/recipes/iris/build.sh
```

Windows looks fine though.

**Comment by ocefpaf** [comment]
I was about to hit the `Comment` button with that message :wink:

File: recipes/iris/build.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    EXT=dylib
+else
+    EXT=so
+fi
+
+SITECFG=lib/iris/etc/site.cfg
+echo "[System]" > $SITECFG
+echo "udunits2_path = $PREFIX/lib/libudunits2.${EXT}" >> $SITECFG
+
+rm -rf lib/iris/tests/results lib/iris/tests/*.npz
```
Comment (pelson) [inline recipes/iris/build.sh]: Incidentally, we don't have the equivalent ```rmdir lib\iris\tests\results /s /q```

File: recipes/iris/build.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
```
Comment (pelson) [inline recipes/iris/build.sh]: eventually I'd like to stop building the unpack extension, and just depend on mo_pack.

**Comment by pelson** [comment]
I cancelled the Windows build. I would happily merge without AppVeyor running over this again if we have a non-intrusive change.

File: recipes/iris/build.sh (hunk 1 above)
Comment (ocefpaf) [inline recipes/iris/build.sh]: Isn't it this line:
https://github.com/conda-forge/staged-recipes/pull/280/files/503a9c95d21cd1356fc0deeb70dd0d1652ef5056#diff-98b76e2d9dcd54f630b042e6a32ca3c9R5

File: recipes/iris/build.sh (hunk 2 above)
Comment (ocefpaf) [inline recipes/iris/build.sh]: Do you want to try that now? Or should we wait to do that in the feedstock?

**Comment by ocefpaf** [comment]
@pelson I am clueless here. No idea how to fix OS X.

File: recipes/iris/build.sh (hunk 1 above)
Comment (pelson) [inline recipes/iris/build.sh]: Sorry, you are right.

Comment (ocefpaf) [inline recipes/iris/build.sh]: You do know that you wrote that line, right? :stuck_out_tongue:

**Comment by ocefpaf** [comment]
Things got worse. Linux was OK before, but now it is segfaulting. OS X is still failing on Python 3.5.

**Comment by ocefpaf** [comment]
I cannot reproduce that `segfault` locally.

**Comment by ocefpaf** [comment]
@pelson isn't it ironic that only Windows is green :stuck_out_tongue_winking_eye:

**Comment by ocefpaf** [comment]
The Linux segfault disappeared as mysteriously as it appeared.

OSX issue now is the mismatch between `hdf5` and `libnetcdf`. Not sure if that is a flaw in our pinning scheme or the same issue we experienced in https://github.com/conda-forge/staged-recipes/pull/444.

File: recipes/iris/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    version: {{ version }}
+
+source:
+    fn: iris-v{{ version }}.tar.gz
+    url: https://github.com/SciTools/iris/archive/v{{ version }}.tar.gz
+    sha256: 93f2cfe1c14aad293103727004863d09f21ea61de337aa73cb96a706df0b6133
+
+build:
+    number: 0
+    skip: True  # [osx]
```
Comment (ocefpaf) [inline recipes/iris/meta.yaml]: @pelson we have two issue on OS X.

- Python 2.7: downloads the most up-to-date `hdf5` and crashes due to a mismatch with `libnetcdf`. I am not sure if that is a bad pinning on our part or the same `conda-build` bug we experienced in RIOS. (Which I believe has a fix in the next relese.)

- Python 3: [`ImportError: No module named 'iris.fileformats._pyke_rules.compiled_krb'`](https://github.com/conda-forge/staged-recipes/pull/280#issuecomment-206377121)


I can confirm that adding the proper `hdf5` in the build requirements the first one, but I don't want to do that as it is a hack. The second one I have no idea. That is why I am skipping OS X here. If you are OK with that let's merge and fix that in the feedstock.

**Comment by ocefpaf** [comment]
@pelson this is the last blocker for IOOS folks to fully adopt `conda-forge`. I am merging this without the OS X builds, but with someone with a Mac can take a look I really appreciate it.

**Pull request merged**